### PR TITLE
haproxy-devel, php7 array usage.. drop init() and !isarray(x)x=array() , instead use getarraybyref() where required.

### PR DESCRIPTION
**Pull request by PiBa-NL**
haproxy-devel, php7 array usage.. drop init() and !isarray(x)x=array() , instead use getarraybyref() where required.

### CODE REVIEW

File: net/pfSense-pkg-haproxy-devel/files/usr/local/pkg/haproxy/haproxy_utils.inc [hunk 1]
```diff
@@ -102,25 +102,6 @@ if(!function_exists('is_arrayset')){
...
-	for($i = 1; $i < count($arg); $i++) {
-		$itemindex = $arg[$i];
-		if (!is_array($item[$itemindex])) {
-			$item[$itemindex] = array();
-		}
-		$item = &$item[$itemindex];
-	}
-	return $item;
-}
-
```
Comment (rbgarga) [inline net/pfSense-pkg-haproxy-devel/files/usr/local/pkg/haproxy/haproxy_utils.inc]: Please don't remove it. Add a `if (!function_exists())` conditional so we can cherry-pick it to 2.4.3 and keep packages the same on both branches as much as possible. It makes our lives easier to maintain packages on branches based on a single version

Comment (PiBa-NL) [reply net/pfSense-pkg-haproxy-devel/files/usr/local/pkg/haproxy/haproxy_utils.inc]: @jim-p & @rbgarga can you decide one way or the other, seems i'm getting mixed messages (or i misinterpreted the intended size of effect).
See same kinda topic which lead me to delete the function here from haproxy as well: https://github.com/pfsense/FreeBSD-ports/pull/553#discussion_r211390033

As for a personal preference, i did like the ```if (!function_exists())``` approach..

Comment (rbgarga) [reply net/pfSense-pkg-haproxy-devel/files/usr/local/pkg/haproxy/haproxy_utils.inc]: Sorry about that @PiBa-NL, we will discuss it and come up with a solution soon.

**Comment by bott0r** [comment]
please approve !

File: net/pfSense-pkg-haproxy-devel/files/usr/local/pkg/haproxy/haproxy_utils.inc (hunk 1 above)
Comment (rbgarga) [reply net/pfSense-pkg-haproxy-devel/files/usr/local/pkg/haproxy/haproxy_utils.inc]: Leave it as is please.

**Pull request merged**